WordPress/WordPressUITests/Utils/XCTest+Extensions.swift

```diff
@@ -52,6 +52,14 @@ extension XCTestCase {
         systemAlertHandler(alertTitle: "“WordPress” Would Like to Access Your Photos", alertButton: "OK")
     }
 
+    public func takeScreenshotOfFailedTest() {
+        if let failureCount = testRun?.failureCount, failureCount > 0 {
```

Comment:
Is this reset after each test? Does it work for multiple failures?

Reply:
> Is this reset after each test?

Yes, this is reset after each test (`testRun` is specific to each test execution); it will only count failures in the test that was just run.

> Does it work for multiple failures?

Our tests are set up to stop after the first failure (`continueAfterFailure = false` in the test setup) so the teardown is done right after a failure — there won't be multiple failures in a test. It wouldn't work the same way if we allowed multiple failures; we'd have to take a different approach in that case. I can see how that check could be misleading if we ever change that setting — I could add a note about it only working in tandem with `continueAfterFailure` set to false.

But this reminds me that it won't screenshot any failures that happen _during_ the teardown (only failures that occur before that point). Do you think that's ok for now or should I continue looking for a broader approach?

Reply:
Ah, that makes sense. I think this is fine as is. If we change `continueAfterFailure` we can reconsider.

Giving it another pass now.